include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl.h

```diff
@@ -2055,6 +2055,202 @@ __parallel_reduce_by_segment(oneapi::dpl::__internal::__device_backend_tag, _Exe
         oneapi::dpl::unseq_backend::__has_known_identity<_BinaryOperator, __val_type>{});
 }
 
+//------------------------------------------------------------------------
+// parallel_scan_by_segment - sync pattern
+//------------------------------------------------------------------------
+template <typename _CustomName, bool __is_inclusive, typename _Range1, typename _Range2, typename _Range3,
+          typename _BinaryPredicate, typename _BinaryOperator, typename _InitType>
+__future<sycl::event, __result_and_scratch_storage<
+                          oneapi::dpl::__internal::tuple<std::uint32_t, oneapi::dpl::__internal::__value_t<_Range2>>>>
```

Comment:
Are you sure about this return type?
For example, `__parallel_transform_reduce_then_scan` return type is `__future<sycl::event, __result_and_scratch_storage<typename _InitType::__value_type>>`
So what is it  in this case?

Reply:
Yes, because we are using the `_WrappedInitType` defined within this function to call transform reduce-then-scan. It ends up being a tuple of the flag and value type that is passed as init to `__parallel_transform_reduce_then_scan` which is reflected in the return type.

Reply:
ok